### PR TITLE
[PW-5424] adyen giving web component

### PR DESCRIPTION
**Pull request by peterojo**
**Description**

Add adyen giving component to regular checkout page for supported payment methods.
- Save `donationToken` from adyen payments response 
- render giving component if enabled and donation token exists
- handle response from onDonate event and show result to shopper

Before giving
Screen Shot 2021-11-02 at 5 31 05 PM
After giving
Screen Shot 2021-11-02 at 5 36 17 PM

**Tested scenarios**

- CC, 3DS1 & 2 success and cancel flows
- Giving component is not rendered if disabled in administration
- Giving component is not rendered for non-supported PMs

### CODE REVIEW

File: Helper/Config.php [hunk 1]
```diff
@@ -212,6 +212,53 @@ public function getPWAOriginUrl($storeId)
         return $this->getConfigData(self::XML_PAYMENT_ORIGIN_URL, self::XML_ADYEN_ABSTRACT_PREFIX, $storeId);
     }
 
+    public function adyenGivingEnabled($storeId)
+    {
+        return $this->getConfigData('active', 'adyen_giving', $storeId);
```
Comment (acampos1916) [inline Helper/Config.php:217]: `adyen_giving` is in a constant right?

File: etc/adminhtml/system/adyen_giving.xml [hunk 2]
```diff
@@ -66,7 +66,7 @@
             <label>Suggested donation amounts</label>
             <backend_model>Adyen\Payment\Model\Config\Backend\DonationAmounts</backend_model>
             <config_path>payment/adyen_giving/donation_amounts</config_path>
-            <tooltip><![CDATA[Suggested amounts that the shopper can choose to donate. The minimum amount is 1 EUR.]]></tooltip>
+            <tooltip><![CDATA[Suggested amounts that the shopper can choose to donate. Amount should be specified in minor units, e.g 100 = 1 EUR. The minimum amount is 1 EUR.]]></tooltip>
```
Comment (acampos1916) [inline etc/adminhtml/system/adyen_giving.xml:69]: I think that asking for minor units here can be confusing, is there something we could do to format it when it is about to be used? Using `\Adyen\Payment\Helper\Data::formatAmount` for example.

File: view/frontend/templates/checkout/success.phtml [hunk 3]
```diff
@@ -45,4 +45,69 @@
...
+<script>
+    <?=$scriptString?>
+    require([
+        'jquery',
+        'Adyen_Payment/js/bundle',
+        'Adyen_Payment/js/model/adyen-payment-service'
+    ], function ($, adyenComponent, adyenPaymentService) {
+        function handleOnDonate(state, component) {
+            state.isValid // True or false. Specifies whether the shopper has selected a donation amount.
+            if (state.isValid) {
```
Comment (acampos1916) [inline view/frontend/templates/checkout/success.phtml:65]: Seems like a typo
```suggestion
            // True or false. Specifies whether the shopper has selected a donation amount.
            if (state.isValid) {
```

**Review by acampos1916 — COMMENTED** [review]
Left a few comments.

File: view/frontend/templates/checkout/success.phtml (hunk 3 above)
Comment (peterojo) [reply view/frontend/templates/checkout/success.phtml]: addressed in 0c4e426

File: etc/adminhtml/system/adyen_giving.xml (hunk 2 above)
Comment (peterojo) [reply etc/adminhtml/system/adyen_giving.xml]: addressed in 0c4e426

File: Helper/Config.php (hunk 1 above)
Comment (peterojo) [reply Helper/Config.php]: addressed in 0c4e426

File: Controller/Process/Result.php [hunk 4]
```diff
@@ -259,6 +264,11 @@ protected function validateResponse($response)
             $this->orderResourceModel->save($order);
         }
 
+        // Save donation token if available in the response
+        if (!empty($response['donationToken'])) {
```
Comment (Morerice) [inline Controller/Process/Result.php:268]: Can you create some ENUMs with the response parameters used?

**Review by Morerice — APPROVED** [review]

**Review by AlexandrosMor — APPROVED** [review]

**Pull request merged**